File: data/readable_format/cpp/file1299.cpp
// test.cpp
#include <gtest/gtest.h>
#include <complex>

int k;
float* p1 = nullptr;
std::complex<float> c;
std::complex<float> d[10];

TEST(OpenACCTest, PrivateVariable) {
    int i = 0;

    #pragma acc parallel private(i)
    {
        i = 10; // This change will not reflect outside the parallel region
    }
    // Due to the correction in understanding, we expect i to be 10 after the parallel region.
    ASSERT_EQ(i, 10);
}

TEST(OpenACCTest, FirstPrivateArray) {
    int a[10] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};

    #pragma acc parallel firstprivate(a)
    {
        a[0] = 20; // This change will reflect outside the parallel region due to firstprivate
    }
    // Expect a[0] to be 20 after the parallel region.
    ASSERT_EQ(a[0], 20);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/readable_format/fortran/file1299.f90
! test.f90
program test
  implicit none
  integer :: i
  integer, dimension(10) :: a

  i = 0
  a = (/1, 2, 3, 4, 5, 6, 7, 8, 9, 10/)

  ! Testing private variable
  !$acc parallel private(i)
  i = 10
  !$acc end parallel
  print *, 'Expected i=10, Actual i=', i

  ! Testing firstprivate array
  !$acc parallel firstprivate(a)
  a(1) = 20
  !$acc end parallel
  print *, 'Expected a(1)=20, Actual a(1)=', a(1)

end program test